abstract_encoder.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod, abstractproperty
+
+class AbstractEncoder(ABC):
+    @abstractmethod
+    def encode(self, text: str):
+        pass
+
+    @abstractmethod
+    def similarity(self, veca, vecb):
```

Comment:
Do we need `similarity` and `average_embedding` to be encoder's methods? 
Don't they look the same for all embeddings?

Reply:
IMO just keep it, maybe give default implementation as in MiniLM